src/libraries/NormalizedAmount.sol

```diff
@@ -1,58 +1,137 @@
+/// @dev NormalizedAmount is a utility library to handle token amounts with different decimals
 // SPDX-License-Identifier: Apache 2
 pragma solidity >=0.8.8 <0.9.0;
 
-type NormalizedAmount is uint64;
+struct NormalizedAmount {
+    uint64 amount;
+    uint8 decimals;
+}
 
-using {gt as >, lt as <, sub as -, add as +, div, min, unwrap} for NormalizedAmount global;
```

Comment:
why remove these? we should keep them, then the downstream code wouldn't have to change at all I think

Reply:
operators can only be defined for user-defined value types. value-types can only be defined for elementary types

Reply:
oh you're right. what a nonsensical restriction...
we could do
```
using {gt, lt, sub, add, div, min, unwrap} for NormalizedAmount global;
```
so we can write `foo.gt(bar)` instead of `NormalizedAmountLib.gt(foo, bar)`

Reply:
i can already do that if the function is part of the library.